content/blog/2024/grid-masonry-nuance.md

```diff
@@ -0,0 +1,480 @@
+---
+title: Choosing a masonry syntax in CSS
+sub: What makes something a 'grid', and what's at stake?
+author: miriam
+date: 2024-09-26
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+Back in 2020,
+Firefox released an experimental `masonry` keyword
+as part of CSS grid layout.
+We can define some grid tracks on one axis,
+set the other grid axis to `masonry`,
+and we get a 'waterfall' of content
+divided somewhat evenly across our tracks --
+aligned on one axis,
+but packing more densely on the other.
+
+<figure>
+  {{ embed.codepen(
+    id='QWjqbJj',
+    title='Masonry Layout Demo',
+    user='jensimmons'
+  ) }}
+  <figcaption>
+    Demo from Jen Simmons works on Safari Tech Preview,
+    or Firefox with a feature flag.
+    At it's core,
+    a 'masonry' layout
+    works like 'grid' layout on one axis
+    and 'flexbox' on the other.
+  </figcaption>
+</figure>
+
+Rachel Andrew immediately
+[pushed back on the proposal](https://rachelandrew.co.uk/archives/2020/05/05/does-masonry-belong-in-the-css-grid-specification/) --
+suggesting that masonry and grid
+are different enough
+they should not be part of the same layout mode.
+Since then, the debate has heated up
+with conflicting proposals from Apple and Google:
+
+- Safari implemented the grid-masonry proposal,
+  and [wrote about the feature](https://webkit.org/blog/15269/help-us-invent-masonry-layouts-for-css-grid-level-3/),
+  [asking for feedback in a CSSWG issue](https://github.com/w3c/csswg-drafts/issues/10233)
+- Chrome responded with an
+  [alternative masonry proposal](https://developer.chrome.com/blog/masonry),
+  and opened an
+  [alternative issue on the CSSWG](https://github.com/w3c/csswg-drafts/issues/9041)
+- For better comparison,
+  two of the primary CSS spec authors --
+  Elika Etemad (Apple) and Tab Atkins-Bittner (Google) --
+  wrote a [combined specification](https://www.w3.org/TR/css-grid-3/)
+  with both proposals side by side
+  (and joined wherever possible)
+- Chrome has since followed-up
+  with [further arguments](https://developer.chrome.com/blog/masonry-syntax)
+  for a non-grid syntax
```

Comment:
```suggestion
- Safari implemented the grid-masonry proposal,
  and [wrote about the feature](https://webkit.org/blog/15269/help-us-invent-masonry-layouts-for-css-grid-level-3/),
  [asking for feedback in a CSSWG issue](https://github.com/w3c/csswg-drafts/issues/10233).
- Chrome responded with an
  [alternative masonry proposal](https://developer.chrome.com/blog/masonry),
  and opened an
  [alternative issue on the CSSWG](https://github.com/w3c/csswg-drafts/issues/9041).
- For better comparison,
  two of the primary CSS spec authors --
  Elika Etemad (Apple) and Tab Atkins-Bittner (Google) --
  wrote a [combined specification](https://www.w3.org/TR/css-grid-3/)
  with both proposals side by side
  (and joined wherever possible).
- Chrome has since followed up
  with [further arguments](https://developer.chrome.com/blog/masonry-syntax)
  for a non-grid syntax.
```